Prevent multiple evaluation of macro argument on ppc.
from OpenBSD ports

also merging this ppc patch from Fedora:
https://src.fedoraproject.org/rpms/libmad/blob/f26/f/libmad-0.15.1b-ppc.patch

--- a/fixed.h
+++ b/fixed.h
@@ -358,15 +358,15 @@
  * significant bit depends on OPT_ACCURACY via mad_f_scale64().
  */
 #  define MAD_F_MLX(hi, lo, x, y)  \
-    do {  \
+   ({ mad_fixed_t __x = (x);  \
+      mad_fixed_t __y = (y);  \
       asm ("mullw %0,%1,%2"  \
 	   : "=r" (lo)  \
-	   : "%r" (x), "r" (y));  \
+	   : "%r" (__x), "r" (__y));  \
       asm ("mulhw %0,%1,%2"  \
 	   : "=r" (hi)  \
-	   : "%r" (x), "r" (y));  \
-    }  \
-    while (0)
+	   : "%r" (__x), "r" (__y));  \
+   })
 
 #  if defined(OPT_ACCURACY)
 /*
@@ -379,8 +379,8 @@
        asm ("addc %0,%2,%3\n\t"  \
 	    "adde %1,%4,%5"  \
 	    : "=r" (lo), "=r" (hi)  \
-	    : "%r" (lo), "r" (__lo),  \
-	      "%r" (hi), "r" (__hi)  \
+	    : "0" (lo), "r" (__lo), \
+	      "1" (hi), "r" (__hi) \
 	    : "xer");  \
     })
 #  endif
